### PR TITLE
Add npm list functionality on publish.

### PR DESCRIPTION
**Pull request by KyleAthom**
Check if all dependencies are present before publishing an app.

### CODE REVIEW

File: lib/App/index.js [hunk 1]
```diff
@@ -267,6 +267,22 @@ class App {
...
 
+    if( await fse.pathExists(path.join(this.path, 'package.json')) ) {
+      const hasAllModules = await NpmCommands.listModules( {path: this.path} );
+      if (!hasAllModules) {
+        const continueOnError = await inquirer.prompt([
+          {
+            type: 'confirm',
+            name: 'value',
+            default: false,
+            message: `Not all node modules have been installed. Are you sure you want to publish this app?`
```
Comment (RobinBol) [inline lib/App/index.js]: "are installed" *

"..., are you sure you want to continue?" *

File: lib/App/index.js [hunk 2]
```diff
@@ -267,6 +267,22 @@ class App {
...
+          },
+        ]);
+        if( !continueOnError.value === true)
```
Comment (RobinBol) [inline lib/App/index.js]: Why not check for `=== false` or `!continueOnError.value`?

File: lib/Modules/NpmCommands.js [hunk 3]
```diff
@@ -93,6 +93,21 @@ class NpmCommands {
...
+    if (!npmInstalled) {
+      Log(colors.red('✖ Could not execute npm list, please install npm globally'));
+      return false; // Return false to be safe
+    }
+
+    try {
+      const { err, stdout } = await exec('npm list', { cwd: path });
+      return true
+    } catch (err) {
+      return false
```
Comment (RobinBol) [inline lib/Modules/NpmCommands.js]: Maybe we want to log the stderr (or err depending on the contents) to give the developer more information on what exactly is missing?

Comment (KyleAthom) [reply lib/Modules/NpmCommands.js]: I thought of that, but npm list produces a lot of output and I didn't want to create a parser for it.

**Pull request merged**